### ISSUE TITLE
SA1011 false positive, C#8.0 when using ! forgiveness operator

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#3052]
```cs
IReadOnlyList<object?>? arguments; // a parameter or assigned elsewhere
object o = arguments[0]!; // SA1011 does not like the `!` so close to the `]` character
```

Similar to #2927
Repros with 1.2.0-beta.66
